src/main/java/io/jenkins/plugins/checks/github/GitHubChecksPublisherFactory.java

```diff
@@ -37,42 +45,67 @@ public GitHubChecksPublisherFactory() {
 
     @Override
     protected Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final TaskListener listener) {
-        return createPublisher(run, DisplayURLProvider.get().getRunURL(run), listener);
-    }
-
-    @VisibleForTesting
-    Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final String runURL, final TaskListener listener) {
-        PluginLogger logger = createLogger(getListener(listener));
-        
-        GitSCMChecksContext gitSCMContext = new GitSCMChecksContext(run, runURL);
-        if (gitSCMContext.isValid(logger)) {
-            return Optional.of(new GitHubChecksPublisher(gitSCMContext, getListener(listener)));
+        try {
+            return createPublisher(run, DisplayURLProvider.get().getRunURL(run), listener);
+        }
+        catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "Could not create logger.", e);
         }
 
-        return createPublisher(listener, logger, new GitHubSCMSourceChecksContext(run, runURL, scmFacade));
+        return Optional.empty();
     }
 
     @Override
     protected Optional<ChecksPublisher> createPublisher(final Job<?, ?> job, final TaskListener listener) {
-        return createPublisher(job, DisplayURLProvider.get().getJobURL(job), listener);
+        try {
+            return createPublisher(job, DisplayURLProvider.get().getJobURL(job), listener);
+        }
+        catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.WARNING, "Could not create logger.", e);
+        }
+
+        return Optional.empty();
     }
 
     @VisibleForTesting
-    Optional<ChecksPublisher> createPublisher(final Job<?, ?> job, final String jobURL, final TaskListener listener) {
-        PluginLogger logger = createLogger(getListener(listener));
+    Optional<ChecksPublisher> createPublisher(final Run<?, ?> run, final String runURL, final TaskListener listener)
+            throws UnsupportedEncodingException {
```

Comment:
what do you expect consumers to do here?
This doesn't seem like it would ever happen.

Better to remove it from the API and handle it internaly to the method I think

Reply:
it is thrown by the constructor of `PrintStream`. Now I'' refactor and use the `FilteredLog` implemented by Ulli in his plugin util, then we don't have to construct those streams and this will be removed.